spec/engines/cached_search_api_connection_spec.rb

```diff
@@ -35,29 +30,20 @@
     end
   end
 
-  describe '#get', vcr: { record: :skip } do
-    context 'on cache hit' do
-      before do
-        expect(cache).to receive(:read).with(endpoint, params).and_return(response)
-      end
+  describe '#get' do
+    context 'when response has not been cached' do
+      it 'calls the api' do
+        expect(cached_connection.connection).to receive(:get).with(endpoint, params)
```

Comment:
Nit-picky since it seems codeclimate isn't complaining about it, but I've generally seen us prefer `have_received` and setting a spy in a before block, over this style of placing `to receive` before the action in the body of the test.  Not a blocking issue at all.

Reply:
I keep getting other errors. I'll leave it as is for now
```
     Failure/Error: expect(connection).to have_receive(:get).with(endpoint, params)

     NoMethodError:
       undefined method `with' for #<RSpec::Matchers::BuiltIn::Has:0x000000011b0a43a0 @method_name=:have_receive, @args=[:get], @block=nil>

               expect(connection).to have_receive(:get).with(endpoint, params)
                                                       ^^^^^
```